lib/Command/InstallModelsCommand.php

```diff
@@ -0,0 +1,207 @@
+ *
+ * @author Matias De lellis <mati86dl@gmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OCA\FaceRecognition\Command;
+
+use OCP\ITempManager;
+use OCP\App\IAppManager;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class InstallModelsCommand extends Command {
+
+	/** @var \OCP\App\IAppManager **/
+	protected $appManager;
+
+	/** @var ITempManager */
+	protected $tempManager;
+
+	/** @var string */
+	protected $tempFolder;
+
+	/** @var string */
+	protected $modelsFolder;
+
+	/* @var  OutputInterface */
+	protected $logger;
+
+	/*
+	 * Models
+	 */
+	private $detectorModelUrl = 'https://github.com/davisking/dlib-models/raw/94cdb1e40b1c29c0bfcaf7355614bfe6da19460e/mmod_human_face_detector.dat.bz2';
+	private $detectorModel = 'mmod_human_face_detector.dat';
+
+	private $resnetModelUrl = 'https://github.com/davisking/dlib-models/raw/2a61575dd45d818271c085ff8cd747613a48f20d/dlib_face_recognition_resnet_model_v1.dat.bz2';
+	private $resnetModel = 'dlib_face_recognition_resnet_model_v1.dat';
+
+	private $predictorModelUrl = 'https://github.com/davisking/dlib-models/raw/4af9b776281dd7d6e2e30d4a2d40458b1e254e40/shape_predictor_5_face_landmarks.dat.bz2';
+	private $predictorModel = 'shape_predictor_5_face_landmarks.dat';
+
+	/**
+	 * @param FaceManagementService $faceManagementService
+	 * @param IUserManager $userManager
+	 */
+	public function __construct(IAppManager  $appManager,
+	                            ITempManager $tempManager) {
+		parent::__construct();
+
+		$this->appManager = $appManager;
+		$this->tempManager = $tempManager;
+	}
+
+	protected function configure() {
+		$this
+			->setName('face:install_models')
+			->setDescription('Download and install Model 1');
+	}
+
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		$this->logger = $output;
+
+		$this->tempFolder = $this->tempManager->getTemporaryFolder('/facerecognition/');
+		$this->modelsFolder = $this->appManager->getAppPath('facerecognition') . '/vendor/models/1/';
```

Comment:
so....how this works? If now are our models *not* part of our install/bundle/zip and they reside *inside* our app, what happens when users upgrade our app? models do disappear and user need to execute `./occ face:install_models` again? If this is not true, ignore my comment. But if this is true, this would be real pain for users. However, I am not sure what other location we can have at our disposal to place models there?

Reply:
> so....how this works? If now are our models not part of our install/bundle/zip and they reside inside our app, what happens when users upgrade our app? models do disappear and user need to execute ./occ face:install_models again?

Wow.. God point..

Reply:
> However, I am not sure what other location we can have at our disposal to place models there?

appdata_INSTANCE_ID seems a good place.. There are saved the preview, avatars, richdocuments templates, deck images, etc. etc.